lambda/shared/src/main/scala/feral/lambda/events/S3Event.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.events
+
+import io.circe.Decoder
+
+import java.time.Instant
+
+final case class S3Event(records: List[S3EventRecord])
+
+object S3Event {
+  implicit val decoder: Decoder[S3Event] = Decoder.forProduct1("Records")(S3Event.apply)
+}
+
+final case class S3EventRecord(
+    eventVersion: String,
+    eventSource: String,
+    awsRegion: String,
+    eventTime: Instant,
+    eventName: String,
+    userIdentity: S3UserIdentity,
+    requestParameters: S3RequestParameters,
+    responseElements: S3ResponseElements,
+    s3: S3,
+    glacierEventData: Option[S3EventRecordGlacierEventData])
+
+object S3EventRecord {
+
+  implicit private val instantDecoder: Decoder[Instant] = Decoder[String].map(Instant.parse(_))
```

Comment:
Can we use this one?
https://github.com/circe/circe/blob/172c9a4d9e14270bab59700d9ffe439462c23db7/modules/core/shared/src/main/scala/io/circe/Decoder.scala#L1241-L1244

Reply:
Probably, looks quite similar. I'll try, the test covers that.

Reply:
Yes, that works. Though, I had to keep the implicit val to prevent `feral.lambda.events.instantDecoder` from being used.

Reply:
Oh I see, there was a conflict with this one 🤔 
https://github.com/typelevel/feral/blob/cbd83badfe6f5af45bc102abe12bc704cb02563a/lambda/shared/src/main/scala/feral/lambda/events/package.scala#L26-L27

Maybe we should make that one not implicit 🤔 anyway, that can be a follow-up PR.

Reply:
#356 could be that follow-up (or preparation) PR for this one.